src/common/tracer.h

```diff
@@ -0,0 +1,66 @@
+#ifndef TRACER_H_
+#define TRACER_H_
+
+#define SIGNED_RIGHT_SHIFT_IS 1
+#define ARITHMETIC_RIGHT_SHIFT 1
+
+#include <iostream>
+#include <arpa/inet.h>
+#include <yaml-cpp/yaml.h>
+
+#include <jaegertracing/Tracer.h>
+
+#ifndef WITH_JAEGER
+  #define WITH_JAEGER
+#endif
+
+typedef std::unique_ptr<opentracing::Span> Span;
+
+class Jager_Tracer{
+  public:
+    Jager_Tracer(){}
+    ~Jager_Tracer(){
+      if(this->tracer == NULL)
+	return;
+      if(!this->isTracerClosed)
+        this->tracer->Close();
+        this->isTracerClosed=true;
+    }
+
+    void init_tracer(const char* tracerName,const char* filePath){
```

Comment:
use ctor for that.
instantiate this class from rgw_main so you can pas conf parameters

Reply:
noted